File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌接口
 * @author Ly
 *
 */
public interface BrandService {

	/**
	 * 查询所有品牌信息
	 * @return 所有品牌信息
	 */
	List<TbBrand> findAll();
	
	
	/**
	 * 分页查询
	 * @param page
	 * @param size
	 * @return
	 */
	PageResult findPage(int page, int size);
	
	/**
	 * 条件查询
	 * @param tbBrand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

	/**
	 * 添加品牌
	 * @param brand
	 */
	void add(TbBrand brand);

	/**
	 * 修改品牌信息
	 * @param tbBrand
	 */
	void update(TbBrand tbBrand);

	/**
	 * 根据id查询
	 * @param id
	 * @return
	 */
	TbBrand findById(long id);

	/**
	 * 根据id批量删除品牌
	 * @param ids
	 */
	void deleteByIds(String[] ids);

	
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
	app.service('brandService',function($http){
		//查询所有品牌信息
		this.findAll=function(){
			return $http.get("../brand/findAll.do");
		}
		//分页查询品牌列表
		this.findPage=function(page,size,searchEntity){
			return $http.post("../brand/findPage.do?page=" + page + "&size=" +size,searchEntity);
		}
		//添加品牌信息
		this.save=function(entity){
			return $http.post("../brand/save.do",entity)
		}
		//修改品牌信息
		this.update=function(entity){
			return $http.post("../brand/update.do",entity);
		}
		//根据id查询品牌
		this.findById=function(id){
			return $http.post("../brand/findById.do?id=" + id);
		}
		//点击删除发送请求
		this.dele=function(selectIds){
			return $http.get("../brand/delete.do?ids=" + selectIds);
		}
		//查条件查询
		this.search=function(page,size,searchEntity){
			return $http.post("../brand/search.do?page=" + page + "&size=" +size,searchEntity);
		}
		
	});